src/effects/backends/audiounit/audiouniteffectprocessor.h

```diff
@@ -70,5 +70,5 @@ class AudioUnitEffectProcessor final : public EffectProcessorImpl<AudioUnitEffec
     void syncParameters();
     void syncStreamFormat(const mixxx::EngineParameters& engineParameters);
 
-    NSView* _Nullable createNativeUI(CGSize size);
+    NSView* _Nullable createNativeUI(CGSize size, QString& outError);
```

Comment:
consider returning a `std::variant<NSView* _Nullable, QString>` (as in for example Rusts `std::Result` type). This would nicely map to `std::expected`.

Reply:
I wasn't sure how well that meshes with the ObjC type, since there's reference counting and stuff involved, but I'll give that a try

Reply:
right... yeah idk enough about that to evaluate how feasible it is. I just assumed C++ semantics hold. If you see more value in a different approach then use that.